Simplenote/Classes/SPNoteListViewController.m

```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
     return NO;
 }
 
-- (UITableViewCellEditingStyle)tableView:(UITableView *)tableView editingStyleForRowAtIndexPath:(NSIndexPath *)indexPath {
+- (NSArray *)tableView:(UITableView*)tableView editActionsForRowAtIndexPath:(nonnull NSIndexPath *)indexPath{
     
-    return UITableViewCellEditingStyleDelete;
-}
-
-- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
+    Note *note = [self.fetchedResultsController objectAtIndexPath:indexPath];
     
-    if (editingStyle == UITableViewCellEditingStyleDelete) {
+    UITableViewRowAction *delete = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Trash-verb", @"Trash (verb) - the action of deleting a note") handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        // Delete something here
+                                        [SPTracker trackListNoteDeleted];
+                                        [[SPObjectManager sharedManager] trashNote:note];
+                                        [[CSSearchableIndex defaultSearchableIndex] deleteSearchableNote:note];
+                                    }];
+    delete.backgroundColor = [UIColor redColor];
+    
+    UITableViewRowAction *restore = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Restore", @"Restore a note from the trash, markking it as undeleted")  handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        [[SPObjectManager sharedManager] restoreNote:note];
+                                        [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
+                                    }];
+    restore.backgroundColor = [UIColor blueColor];
```

Comment:
Maybe change this one to `orangeColor`? (Restore action in blue, while the permanently delete is in read, looks honestly a bit odd!). Just borrowing color ideas from Apple Mail.app, which displays actions in Gray / Orange / Red.